app/renderer/src/actions/Inspector.js

```diff
@@ -849,22 +879,72 @@ export function setAwaitingMjpegStream(isAwaiting) {
   };
 }
 
+export function uploadGesturesFromFile(fileList) {
+  return async (dispatch) => {
+    const fileReaderPromise = fileList.map((file) => {
+      const reader = new FileReader();
+      return new Promise((resolve) => {
+        reader.onload = (event) =>
```

Comment:
how do we handle possible failures here?

Reply:
I have added the error callbacks to prevent unexpected failures during file upload.